Documentation/developers/hints.rst

```diff
@@ -1,5 +1,90 @@
-Some hints for programmers
---------------------------
+Hints for programmers
+---------------------
+
+Compiling a Polycube service
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Polycube services can be compiled in two ways:
+
+- together with the entire Polycube source code
+- as stand-alone services that can be loaded into Polycube at runtime
+
+Compiling together with the Polycube source code
+************************************************
+1. Place the service into the :SCM_WEB:`src/services <src/services>` folder;
+
+2. Update the :SCM_WEB:`src/services/CMakeLists.txt <src/services/CMakeLists.txt>` file adding a new `cmake` instruction:
+  
+    ::
+
+      add_service(service_name service_folder)
+
+where the first argument is the service name used in the Polycube REST API, while second argument is the folder name of the service.
+
+**Note**: the service folder name must follow the ``pcn-service_name`` naming convention (E.g. pcn-bridge, pcn-router).
+
+3. Compile and re-install Polycube:
+    
+  ::
+
+    # having Polycube root folder as working directory
+    mkdir build && cd build
+    cmake ..
+    make -j $(getconf _NPROCESSORS_ONLN)
+    sudo make install
+
+.. _stand_alone_compilation:
+
+Compiling a stand-alone service
+*******************************
+1. create a `build` folder in the root folder of the service and set it as working directory:
+
+  ::
+
+    mkdir build && cd build
+
+2. generate a Makefile:
+
+  ::
+
+    cmake ..
+
+3. compile the service to build a shared library object:
+
+  ::
+
+    make -j $(getconf _NPROCESSORS_ONLN)
+
+At this point the service has been compiled and the built shared library object can be found into the ``build/src`` folder named as ``libpcn-service_name.so``
```

Comment:
Hi @michelsciortino , thanks for this PR. I would add here a sentence like to better connect the compilation part with the following part of the document: "Your cube, completely encapsulated in the .so file, is now ready to be loaded by the Polycube"